### ISSUE TITLE
bug: app crashes when 'hide account menu' option is enabled and some custom filters are added

### ISSUE THREAD

**Issue by madhurvarshney** [inotia00/ReVanced_Extended#549]
### Type

Error at runtime

### Application

YouTube v18.11.35

### Bug description

App crashes when playing any video from History section, with 'hide account menu' option ON and some additional custom filters are added.

### Steps to reproduce

1. Enable 'hide account menu' option.
2. Add some additional custom filters, like 'Your movies,Get YouTube Premium,Help and feedback'.
3. Restart the app.
4. Play any video from 'History' section.
5. Swipe down/press back to close that video.
6. Again play any video from the 'History' section.
7. Swipe down/press back.
8. App crashes.

### Relevant log output

```shell
2023-03-24 11:09:08.290 23595-23595 AndroidRuntime          app.rvx.android.youtube              E  FATAL EXCEPTION: main
                                                                                                    Process: app.rvx.android.youtube, PID: 23595
                                                                                                    java.lang.ClassCastException: android.widget.LinearLayout$LayoutParams cannot be cast to nx
                                                                                                    	at android.support.v7.widget.RecyclerView.l(PG:1)
                                                                                                    	at android.support.v7.widget.RecyclerView.aQ(PG:2)
                                                                                                    	at android.support.v7.widget.RecyclerView.D(PG:86)
                                                                                                    	at android.support.v7.widget.RecyclerView.onLayout(PG:1)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(PG:9)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at aege.to(PG:19)
                                                                                                    	at aegg.h(PG:1)
                                                                                                    	at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.h(PG:1)
                                                                                                    	at com.google.android.apps.youtube.app.ui.actionbar.MainScrollingViewBehavior.h(PG:1)
                                                                                                    	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(PG:6)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at com.google.android.apps.youtube.app.watch.nextgenwatch.ui.NextGenWatchContainerLayout.g(PG:2)
                                                                                                    	at com.google.android.apps.youtube.app.watch.nextgenwatch.ui.NextGenWatchContainerLayout.h(PG:7)
                                                                                                    	at com.google.android.apps.youtube.app.watch.nextgenwatch.ui.NextGenWatchContainerLayout.onLayout(PG:4)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at com.google.android.apps.youtube.app.common.ui.navigationbar.NavigationBarDividerLayout.onLayout(PG:4)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at com.google.android.apps.youtube.app.ui.swipetocontainer.SwipeToContainerFrameLayout.onLayout(PG:1)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at bfa.onLayout(PG:6)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1841)
                                                                                                    	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
                                                                                                    	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1841)
                                                                                                    	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
                                                                                                    	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
2023-03-24 11:09:08.290 23595-23595 AndroidRuntime          app.rvx.android.youtube              E  	at com.android.internal.policy.DecorView.onLayout(DecorView.java:872)
                                                                                                    	at android.view.View.layout(View.java:23963)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6506)
                                                                                                    	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4142)
                                                                                                    	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3485)
                                                                                                    	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2442)
                                                                                                    	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9375)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1388)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1396)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1033)
                                                                                                    	at android.view.ChoreographerExtImpl.checkScrollOptSceneEnable(ChoreographerExtImpl.java:408)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:900)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1371)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:942)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:240)
                                                                                                    	at android.os.Looper.loop(Looper.java:351)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8364)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
```

### Screenshots or videos

https://user-images.githubusercontent.com/125809686/227311953-0ae121ba-14de-49d9-a1ba-446006f0d0f6.mp4

### Device Environment

Android 13, Realme UI 4

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by inotia00** [comment]
please attach crash log

**Comment by madhurvarshney** [comment]
> please attach crash log

just check if the crash log I have attached is relevant?

**Comment by inotia00** [comment]
oh, i checked now

btw seems like a totally unrelated error
(An error occurred in a class unrelated to the location where the patch was made)

Is it possible to check if the issue reproduces when you exclude the `hide-account-menu` patch?

**Comment by madhurvarshney** [comment]
I checked by excluding the `hide-account-menu` patch and the issue does not happen then.

Some more info:
Even if I include the `hide-account-menu` patch, the issue is not encountered in every case. I tried every combination of custom filters and found that the issue only occurs when "Get YouTube Premium" or "Help and feedback" or both are added along with "Your movies" filter.
For example, "Your movies,Get YouTube Premium" or "Your movies,Help and feedback" or "Your movies,Get YouTube Premium,Help and feedback" or "YouTube Music,YouTube Kids,Your movies,Get YouTube Premium,Help and feedback" etc.

For any other combinations (like "Get YouTube Premium,Help and feedback" or "Your movies, YouTube Kids" etc.) the issue does not occur.

**Comment by inotia00** [comment]
The issue did not reproduce when I tested it.

Looking at the attached screen record, it seems that the new layout has been applied, so it seems that fixing is possible only when the layout is applied to me.

Anyway thanks for reporting the specific issue, I will keep this issue for archival purposes as it is a useful resource.

**Comment by inotia00** [comment]
Found the cause of the crash and this issue can be fixed

but this setting won't be available in landscape mode (or tablet) if the fix is applied

This is because YouTube uses a hardcoded layout param in landscape mode, so it cannot be implemented as a patch

**Comment by inotia00** [comment]
fixed in revanced-patches-v2.184.3, but still need tests

**Comment by inotia00** [comment]
I have confirmed that it was fix in new layout
closing the issue
